react/src/components/forms/InputTextFuzzy/index.js

```diff
@@ -67,19 +67,23 @@ class InputTextFuzzy extends React.Component {
   // handleChange and onSuggestionSelected both do not fire when enter is hit.
   // This is a workaround for that. Use handleChange for keyboard presses.
   handleKeyPress = (event) => {
-    const { value } = this.state;
+    const { value, suggestions } = this.state;
+    if (is.fn(this.props.onKeyDown)) {
+      this.props.onKeyDown(event);
+    }
+    const match = suggestions.find((el) => el.item.text === value);
     if (event.key === 'Enter') {
       event.persist();
       event.preventDefault();
-      if (is.fn(this.props.onSuggestionClick)) {
+      if (is.fn(this.props.onSuggestionClick) && match) {
         this.props.onSuggestionClick(event, { method: 'enter', suggestion: { item: { text: value } } });
       }
     }
   }
   handleFocus = (event) => {
     if (is.fn(this.props.onFocus)) {
       event.persist();
-      this.props.onFocus({ event, value: this.state.value, suggestions: this.state.suggestions });
+      this.props.onFocus(event, { value: this.state.value, suggestions: this.state.suggestions });
```

Comment:
This is a breaking change to any projects using InputTextFuzzy that relies on event. For now, it might be best to do:
```suggestion
      this.props.onFocus(event, { event, value: this.state.value, suggestions: this.state.suggestions });
```